feat(na): clarify when to use docker vs k8s, rename docker page

mrjones-plip opened the pull request:
<!--
Please use semantic PR titles that respect this format:

<type>(#<issue number>): <subject>

Quick example:

feat(#1234): add hat wobble
^--^(#^--^): ^------------^
|     |      |
|     |      + - > subject
|     |
|     + -------- > issue number
|
+ -------------- > type: chore, feat, fix, perf.

https://docs.communityhealthtoolkit.org/contribute/code/workflow/#commit-message-format
-->

# Description

This PR:
* acknowledges recent forum post [citing shards vs cores findings](https://forum.communityhealthtoolkit.org/t/investigate-adding-more-shards-as-a-potential-avenue-for-improved-performance/4831)
* formally recommends docker for most production hosting 
* calls out k8s hosting is only for either large multi-tentant deployments or deployments with bare-metal servers with 8 or fewer cores
* removes the [horizontal vs vertical scaling page](https://docs.communityhealthtoolkit.org/hosting/vertical-vs-horizontal/) - this is only slightly relevant and just adds confusion as to how the CHT should be hosted. 
* removes the [considerations hosting page](https://docs.communityhealthtoolkit.org/hosting/4.x/production/considerations/).  This is now covered by the k8s vs docker page.
* renames the title of the docker install page from "[Prerequisites](https://docs.communityhealthtoolkit.org/hosting/4.x/production/docker/prerequisites/)" to "Installation"

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

mrjones-plip commented:
cc @Hareet and @andrablaj

dianabarsan commented on content/en/hosting/costs.md:
```diff
@@ -25,9 +25,9 @@ In a production environment, the CHT costs USD$0.10 per active user per month to
 Medic hosts a number of production CHT instances in Amazon [Elastic Kubernetes Service](https://aws.amazon.com/eks/) (EKS). By using [OpenCost](https://www.opencost.io/), Medic can closely monitor real world costs with actual end users. OpenCost was used to separate one deployment's CPU, RAM and disk use from another's despite them all running in a multi-tenant EKS cluster.
 
-By looking at dozens of instances and focusing on the highest usage ones, an accurate representation can be derived of a higher cost a self hosting partner.  
+By looking at dozens of instances and focusing on the highest usage ones, an accurate representation can by a self hosting partner to plan their own hosting.  
```
> an accurate representation can by a self hosting partner to plan their own hosting.

I'm not sure what you meant to say here.

mrjones-plip replied:
yikes!  fixed in c06d8c2a

dianabarsan commented on content/en/hosting/kubernetes-vs-docker.md:
```diff
@@ -2,48 +2,64 @@
+* A CHT API service.
+* HAProxy Healthcheck service.
+* HAProxy service.
+* Upgrade Service service.
```
> Upgrade Service service.

haha

mrjones-plip replied:
:sweat_smile:

dianabarsan commented on content/en/hosting/kubernetes-vs-docker.md:
```diff
@@ -2,48 +2,64 @@
+## Kubernetes (k8s)
+
+{{< callout type="warning" >}}
+Only use Kubernetes on bare-metal machines with 8 cores or fewer or when dozens of CHT instances are needed in a multi-tenant deployment.
```
> bare-metal machines with 8 cores or fewer

I'm not sure we should stipulate the number of cores here. Maybe say low-end instead?

andrablaj replied:
+1 on Diana's comment.

Additionally, I am reluctant of the use of the word "only". We can do recommendations according to deployment needs, but we shouldn't impose a tool or another.

mrjones-plip replied:
gotcha.  Lemme know if e7ad45ba doesn't address your feedback!

dianabarsan reviewed (commented):
Very nice and quick! I'm positively pleasantly surprised :) thank you! I left two minor comments.

andrablaj commented on content/en/hosting/4.x/docker/installation.md:
```diff
@@ -1,15 +1,12 @@
+  - /hosting/4.x/docker/prerequisites/
 ---
 
-{{< hextra/hero-subtitle >}}
```
Can we keep the page description for consistency with the other pages, and change it to something like "Guide for running production with Docker"

mrjones-plip replied:
yah! fixed in 883467d9

andrablaj commented on content/en/hosting/4.x/_index.md:
```diff
@@ -12,9 +12,9 @@ aliases:
 > To get an overview on how these hosting solutions use `docker` and other key CHT concepts, be sure to read the [guide on a Local Setup]({{< relref "building/local-setup" >}}).
 
-Before beginning any of these guides, be sure to meet all of the [CHT hosting requirements]({{< relref "hosting/requirements" >}}) first.
+Before beginning any of these guides, be sure to meet all of the [CHT hosting requirements]({{< relref "hosting/requirements" >}}) first. For example, [backups](/hosting/4.x/docker/backups) are required for any successful CHT deployment!
```
```suggestion
Before beginning any of these guides, be sure to meet all of the [CHT hosting requirements]({{< relref "hosting/requirements" >}}) first. For example, [backups](/hosting/4.x/docker/backups) are required for any successful CHT deployment.
```

andrablaj replied:
nitpick: exclamation marks seem unnecessary in a guide.

andrablaj commented on content/en/hosting/4.x/docker/_index.md:
```diff
@@ -0,0 +1,17 @@
+---
+
+{{< hextra/hero-subtitle >}}
+  Production Hosting CHT 4.x in Docker on a Single CouchDB Node
```
```suggestion
  Production hosting CHT 4.x in Docker on a single CouchDB node
```

andrablaj replied:
to match style on other page descriptions

andrablaj commented on content/en/hosting/4.x/kubernetes/_index.md:
```diff
@@ -0,0 +1,16 @@
+{{< /hextra/hero-subtitle >}}
+
+{{< callout type="warning" >}}
+Only use Kubernetes under [very specific](/hosting/kubernetes-vs-docker/) conditions. [Docker](/hosting/4.x/docker/) is recommended for most deployments.
```
Considering that we encourage large-scale, national deployments and the biggest and most impactful by numbers CHT deployments use Kubernetes, how accurate is this sentence?

mrjones-plip replied:
I've removed `only` from this sentence in a9ebca9c 

We are changing what we recommend.  Now, we encourage large-scale national deployments (ie 10k+ CHWs) to run on Docker - this sentence is accurate! 

For instances that have dozens of deployments where they need to provision a multi-tenant deployment, we encourage k8s for ease of management.

andrablaj commented on content/en/hosting/kubernetes-vs-docker.md:
```diff
@@ -2,48 +2,64 @@
 {{< /hextra/hero-subtitle >}}
 
-Since the release of CHT Core 4.0.0 in late 2022, Medic has been perfecting the hosting for the toolkit to balance the need for high uptimes so CHWs can always deliver care while having an easy and approachable technical back end hosting solution.  While initially [Docker Compose](https://docs.docker.com/compose/) with an [overlay network](https://docs.docker.com/compose/networking/#multi-host-ne ...
+Medic recommends [Docker](/hosting/4.x/docker/) for most deployments. Only use [Kubernetes](/hosting/4.x/kubernetes/) for large, multi-tenant or specific hardware constrained deployments.
```
```suggestion
Medic recommends [Docker](/hosting/4.x/docker/) for small scale deployments. Use [Kubernetes](/hosting/4.x/kubernetes/) for large, multi-tenant or specific hardware constrained deployments.
```

andrablaj replied:
Your sentence assumes most of CHT deployments are small scale deployments which I am unsure is accurate.

mrjones-plip replied:
My sentence actually does not assume most CHT deployments are small scale (eg only 100s of CHWs).  We have proven that Docker can support large scale deployments.  We know that all CHT deployments that are this shape can be run Docker.  Removing "only" here I think softens it enough.

fixed in 65ee12f7

andrablaj commented on content/en/hosting/kubernetes-vs-docker.md:
```diff
@@ -2,48 +2,64 @@
+## Kubernetes (k8s)
+
+{{< callout type="warning" >}}
+Only use Kubernetes on bare-metal machines with 8 cores or fewer or when dozens of CHT instances are needed in a multi-tenant deployment.
```
```suggestion
Use Kubernetes on bare-metal machines with 8 cores or fewer or when dozens of CHT instances are needed in a multi-tenant deployment.
```

mrjones-plip replied:
"only" was removed in another commit.

andrablaj commented on content/en/hosting/kubernetes-vs-docker.md:
```diff
@@ -2,48 +2,64 @@
+Only use Kubernetes on bare-metal machines with 8 cores or fewer or when dozens of CHT instances are needed in a multi-tenant deployment.
+{{< /callout >}}
+
+Given the additional knowledge needed to run and complexities to deploy, CHT on k8s has been shown to result in confusion and outages. Docker is the preferred solution for most production deployments.
```
```suggestion
Running CHT on k8s requires additional knowledge and can be more complex to deploy. Consider using Docker for small scale CHT deployments. 
```

mrjones-plip replied:
Keeping half the suggestion, see fb2b2d68

andrablaj commented on content/en/hosting/kubernetes-vs-docker.md:
```diff
@@ -2,48 +2,64 @@
+
+Given the additional knowledge needed to run and complexities to deploy, CHT on k8s has been shown to result in confusion and outages. Docker is the preferred solution for most production deployments.
+
+K8s deployments can not safely use the one click upgrade button in the admin GUI.
```
Do you have a link to forum posts or data behind this statement to explain the why and what happens if people use that button? I think it will help understand what's lacking currently to have this possible.

mrjones-plip replied:
Yeah - good point!  I think this can actually be addressed (if you back up your helm charts and always use helm charts deply or restore, it works fine).  I'll just remove this for now.

fixed in 19281800

andrablaj commented on content/en/hosting/kubernetes-vs-docker.md:
```diff
@@ -2,48 +2,64 @@
+
+K8s deployments can not safely use the one click upgrade button in the admin GUI.
+
+Otherwise, k8s offers:
```
```suggestion
k8s offers:
```

andrablaj reviewed (changes requested):
Thanks for starting this @mrjones-plip. I appreciate the intention of bringing clarity to the community on hosting options. 

You mentioned the [considerations](https://docs.communityhealthtoolkit.org/hosting/4.x/production/considerations/) page is now covered by the k8s vs docker page. I really loved the simple table with clear data on when to choose one setup or another. Can we have it back to explain what a small-scale deployment is vs a "large, multi-tenant" deployment? users, doc, etc? I think that can help programs make decisions on which tool to use. 

I am also unsure why we removed the [vertical vs horizontal](https://docs.communityhealthtoolkit.org/hosting/vertical-vs-horizontal/) page as that explained quite well when "it's time to horizontally scale". If you feel there are sentences which are not relevant anymore, we can adjust them, but can you please explain why the community would find this page confusing? cc @dianabarsan in case you have additional thoughts. I am definitely not an expert, and just trying to understand what's the most helpful for the readers. 

I understand the need to recommend the simplest solution according to needs, but I would encourage us to share clear numbers to back recommendations and avoid "only use this or that" without saying why.

mrjones-plip commented:
Thanks again for all your input @andrablaj .  I'm gonna go over this whole PR one more time to see about incorporating Hareet's feedback (see below)

----

> I am also unsure why we removed the [vertical vs horizontal](https://docs.communityhealthtoolkit.org/hosting/vertical-vs-horizontal/) page as that explained quite well when "it's time to horizontally scale".

This is a big deal:  I'm intentionally polarizing our documentation to say: "we no longer recommend muliti-node couch" ! This because of @dianabarsan's [findings](https://forum.communityhealthtoolkit.org/t/investigate-adding-more-shards-as-a-potential-avenue-for-improved-performance/4831/3?u=mrjones):

> In today’s world of virtualization, the cost of having 32 cores is roughly the same if you have them in a single machine vs 4 machines, each with 8 cores (at least, that’s how it seems for AWS), so the only consideration becomes performance.

As such, it is no longer relevant to talk about vertical vs horizontal scaling, hence why I deleted this page.  

> You mentioned the [considerations](https://docs.communityhealthtoolkit.org/hosting/4.x/production/considerations/) page is now covered by the k8s vs docker page. I really loved the simple table with clear data on when to choose one setup or another. Can we have it back to explain what a small-scale deployment is vs a "large, multi-tenant" deployment? users, doc, instances etc?

This is a big deal: Docker can handle large scale deployments (lots and LOTS of CHWs)!  This is a change in what we were saying before.  I think removing the page is a good thing and I'd like to have everything we need on the Docker vs k8s page.  If folks think a talbe is a good thing, I can look into adding a different version back. 

-----

from Hareet for mjones to think over some:

>> Use Docker for small-scale deployments and Kubernetes for large-scale deployments (with a link to a paragraph explaining small/large means in practice, e.g. <5 or >=5 instances)
> 
> Hm, I think that's close.  Largely the differentiators are:
>
> Running a variety of services across distributed systems (Clustering would fall under here)
> Multiple instances per admin
> Pre-existing cluster, should also put small-scale projects there for ease of administration
>
> We can decouple it from our tech debt choices (cluster for performance) and go back to "it's a container orchestrator".If you have lots of containers, and need to orchestrate them, with few admins, regardless of their project size, perhaps not just cht, it is far easier to use an orchestrator and we have tutorials around the main ones.

mrjones-plip commented:
@andrablaj and @Hareet and @antonykhaemba - I've added:
* an examples section per the Stewardship call today
* an intro that says `Medic previously suggest Kubernetes for all production deployments - this is no longer the case.`
* a qualifier to use what's most comfy. While we recommend k8s for multi-tenant: `deployments should use the technology they know best. If they plan for 20 instances but really doen't want to use Kubernetes, that's fine.  Conversely, if an NGO only uses Kubernetes and wants to deploy a single instance of the CHT - that's also fine.`

Feedback welcome!

andrablaj commented on content/en/hosting/kubernetes-vs-docker.md:
```diff
@@ -2,48 +2,258 @@
-* Upgrade Service service.
-* A CHT Sentinel service.
-* An nginx service to act as a reverse proxy and terminate TLS connections
+### MoH Kenya
```
@mrjones-plip let's not use the name of the deployment before asking for explicit consent from the MoHes. I suggest we use `Example 1`, `Example 2` instead. Of course, this is also now in the version control of this repo, but we should definitely not use the real deployment names.

mrjones-plip replied:
gotcha - fixed in 6387fe8f

andrablaj commented on content/en/hosting/kubernetes-vs-docker.md:
```diff
@@ -2,48 +2,258 @@
+|            Users: | 25.7k                                           |
+
+
+<!-- 
```
@mrjones-plip, if I'm not mistaken, this is not public data, so we shouldn't add it to our public documentation repository. Please remove it from this file until we receive explicit consent from MoHes to publish information about their deployments.

andrablaj reviewed (changes requested):
I haven't reviewed the rest of the documentation, and I will review them by tomorrow. Meanwhile, please remove any mention of MoHes and related URLs from this PR unless we have explicit written consent. Thank you!

andrablaj commented on content/en/hosting/kubernetes-vs-docker.md:
```diff
@@ -2,48 +2,118 @@
 
-Since the release of CHT Core 4.0.0 in late 2022, Medic has been perfecting the hosting for the toolkit to balance the need for high uptimes so CHWs can always deliver care while having an easy and approachable technical back end hosting solution.  While initially [Docker Compose](https://docs.docker.com/compose/) with an [overlay network](https://docs.docker.com/compose/networking/#multi-host-ne ...
+{{< callout >}}
+Medic previously suggest Kubernetes for all production deployments - this is no longer the case.  Further, Medic no longer suggests multi-node CouchDB.
```
```suggestion
For older versions of the CHT, Medic recommended Kubernetes for all production deployments. This is no longer the recommendation for newer versions of the CHT. Further, multi-node CouchDB is no longer recommended.
```

andrablaj commented on content/en/hosting/kubernetes-vs-docker.md:
```diff
@@ -2,48 +2,118 @@
-* New, smaller production 4.x CHT deployments can use either [Docker]({{< relref "hosting/4.x/production/docker/" >}}) or [Kubernetes]({{< relref "hosting/4.x/production/kubernetes/" >}}) depending on their expected use and growth.
-* Application development for both [CHT 3.x]({{< relref "hosting/3.x/app-developer" >}}) and [CHT 4.x]({{< relref "hosting/4.x/app-developer" >}}) should use Docker Compose.
-* Production 3.x CHT deployments should use [Docker Compose]({{< relref "hosting/3.x" >}})  - Note that 3.x is [end of life]({{< relref "releases/#supported-versions" >}})  and should only be used to support existing 3.x deployments.
+Medic now recommends [Docker](/hosting/4.x/docker/) for most deployments. Use [Kubernetes](/hosting/4.x/kubernetes/) for multi-tenant or specific hardware constrained deployments. For more details on the research behind suggesting Docker and single-node CouchDB, please see [this forum post](https://forum.communityhealthtoolkit.org/t/investigate-adding-more-shards-as-a-potential-avenue-for-improve ...
```
```suggestion
Medic recommends [Docker](/hosting/4.x/docker/) for most deployments. Use [Kubernetes](/hosting/4.x/kubernetes/) for multi-tenant or specific hardware constrained deployments. For more details on the research behind suggesting Docker and single-node CouchDB, see [this forum post](https://forum.communityhealthtoolkit.org/t/investigate-adding-more-shards-as-a-potential-avenue-for-improved-performance/4831?u=mrjones).
```

andrablaj replied:
Per the [docs style guide](https://docs.communityhealthtoolkit.org/community/contributing/docs/style-guide/#simple-and-direct-language), we should avoid please.

mrjones-plip replied:
Thanks for the attention to detail - I'll revisit that style guide as a refresher!

andrablaj commented on content/en/hosting/kubernetes-vs-docker.md:
```diff
@@ -2,48 +2,118 @@
+Medic now recommends [Docker](/hosting/4.x/docker/) for most deployments. Use [Kubernetes](/hosting/4.x/kubernetes/) for multi-tenant or specific hardware constrained deployments. For more details on the research behind suggesting Docker and single-node CouchDB, please see [this forum post](https://forum.communityhealthtoolkit.org/t/investigate-adding-more-shards-as-a-potential-avenue-for-improve ...
 
-## Kubernetes
+That said, deployments should use the technology they know best. If they plan for 20 instances but really doen't want to use Kubernetes, that's fine.  Conversely, if an NGO only uses Kubernetes and wants to deploy a single instance of the CHT - that's also fine.
```
```suggestion
However, deployments should use the technology they are most familiar and comfortable with. If they plan for 20 CHT instances but don't want to use Kubernetes, that's fine. Conversely, if a hosting organisation uses Kubernetes and intends to deploy a single instance of the CHT, that's also fine.
```

andrablaj commented on content/en/hosting/kubernetes-vs-docker.md:
```diff
@@ -2,48 +2,118 @@
-Like Kubernetes above, Docker Compose deploys the same services but adds an additional 7th to allow for ingress/egress via a reverse proxy:
+Below are examples from 2 real world Ministries of Health (MoH) which have been anonymized and Medic hosted instances.  Users are total provisioned users, fewer users may be active in a given month.
+
+### MoH Example 1
```
I appreciate how illustrative the real-world examples are, thank you!

andrablaj reviewed (changes requested):
Thanks for the updates @mrjones-plip, this is looking way clearer now!  

Additionally, please update `hosting/4.x/` [cards](https://docs.communityhealthtoolkit.org/hosting/4.x/) to separate Production Docker and Production Kubernetes.

mrjones-plip commented:
Thanks for the review Andra!  I've added cards in 1826b5e0

andrablaj commented on content/en/hosting/costs.md:
```diff
@@ -25,9 +25,9 @@ In a production environment, the CHT costs USD$0.10 per active user per month to
 Medic hosts a number of production CHT instances in Amazon [Elastic Kubernetes Service](https://aws.amazon.com/eks/) (EKS). By using [OpenCost](https://www.opencost.io/), Medic can closely monitor real world costs with actual end users. OpenCost was used to separate one deployment's CPU, RAM and disk use from another's despite them all running in a multi-tenant EKS cluster.
 
-By looking at dozens of instances and focusing on the highest usage ones, an accurate representation can be derived of a higher cost a self hosting partner.  
+By looking at dozens of instances and focusing on the highest usage ones, hosting partners can know our numbers are applicable.  
```
nitpick: I am unsure what this sentence means, to be honest. 😅 Is there a way to use simpler language, like: The numbers below are backed by the existing CHT deployments.

mrjones-plip replied:
you and Diana were both confused by this - let's yank it! surrounding text is clarifying enough.  see 6550fd90

andrablaj commented on content/en/hosting/costs.md:
```diff
@@ -66,13 +66,13 @@ A production deployment means the CHT is hosted either at a cloud provider or in
-It should be noted that a [development environment]({{< relref "hosting/4.x/app-developer" >}})  can be no cost (or very low cost).  Assuming a developer already has a laptop, this is all that is needed to host a development instance. 
-
-Running an easy to set up Docker based development environment (instead of Kubernetes) on a laptop is an acceptable practice: there is no expectation of high uptime for development instances.
+It should be noted that a [development environment](/hosting/4.x/app-developer)  can be no cost (or very low cost).  Assuming a developer already has a laptop, this is all that is needed to host a development instance. 
```
```suggestion
> [!NOTE]
> A [development environment](/hosting/4.x/app-developer)  can be no cost (or very low cost).  Assuming a developer already has a laptop, this is all that is needed to host a development instance. 
```

andrablaj commented on content/en/hosting/costs.md:
```diff
@@ -66,13 +66,13 @@ A production deployment means the CHT is hosted either at a cloud provider or in
-In order to get a better idea of fixed monthly costs, let's look at the smallest Kubernetes deployment per the [requirements docs]({{< relref "hosting/requirements" >}}).  These estimates use Amazon's [EC2 pricing](https://aws.amazon.com/ec2/pricing/on-demand/) and [EBS pricing](https://aws.amazon.com/ebs/pricing/) in Paris (`eu-west2`) availability zone.  Note that different [types of EC2 pricin ...
+In order to get a better idea of fixed monthly costs, let's look at the smallest Kubernetes deployment per the [requirements docs](/hosting/requirements}).  These estimates use Amazon's [EC2 pricing](https://aws.amazon.com/ec2/pricing/on-demand/) and [EBS pricing](https://aws.amazon.com/ebs/pricing/) in Paris (`eu-west2`) availability zone.  Note that different [types of EC2 pricing](https://aws. ...
+
+For more examples of deployments, but without costs, see [Kubernetes vs Docker](/hosting/kubernetes-vs-docker/#example-deployments).
```
```suggestion
{{< callout >}}
For more examples of deployments, but without costs, see [Kubernetes vs Docker](/hosting/kubernetes-vs-docker/#example-deployments).
{{< /callout >}}
```

andrablaj commented on content/en/hosting/kubernetes-vs-docker.md:
```diff
@@ -2,48 +2,118 @@
+However, deployments should use the technology they are most familiar and comfortable with. If they plan for 20 CHT instances but don't want to use Kubernetes, that's fine. Conversely, if a hosting organisation uses Kubernetes and intends to deploy a single instance of the CHT, that's also fine.
 
-Kubernetes provides advantages in managing CHT Deployments:
+Application development for both [CHT 3.x]({{< relref "hosting/3.x/app-developer" >}}) and [CHT 4.x]({{< relref "hosting/4.x/app-developer" >}}) should use Docker.
```
```suggestion
{{< callout >}}
 Application development for both [CHT 3.x]({{< relref "hosting/3.x/app-developer" >}}) and [CHT 4.x]({{< relref "hosting/4.x/app-developer" >}}) should use Docker.
{{< /callout >}}
```

andrablaj commented on content/en/hosting/kubernetes-vs-docker.md:
```diff
@@ -2,48 +2,118 @@
+## Example deployments
 
-Like Kubernetes above, Docker Compose deploys the same services but adds an additional 7th to allow for ingress/egress via a reverse proxy:
+Below are examples from 2 real world Ministries of Health (MoH) which have been anonymized and Medic hosted instances.  Users are total provisioned users, fewer users may be active in a given month.
```
```suggestion
Below are examples from real world large-scale CHT deployments which have been anonymised.  Users are total provisioned users, fewer users may be active in a given month.
```

andrablaj replied:
I don't see any benefit in mentioning whether Medic is hosted or not. The important information here is the scale and technology used.

dianabarsan replied:
Can we use average monthly active users instead of provisioned users?

mrjones-plip replied:
oop - missed Diana's request for average monthly active users - unresolving!

mrjones-plip replied:
moved all instances to 30 day average active user  in 986f4df8

andrablaj commented on content/en/hosting/kubernetes-vs-docker.md:
```diff
@@ -2,48 +2,118 @@
-Like Kubernetes above, Docker Compose deploys the same services but adds an additional 7th to allow for ingress/egress via a reverse proxy:
+Below are examples from 2 real world Ministries of Health (MoH) which have been anonymized and Medic hosted instances.  Users are total provisioned users, fewer users may be active in a given month.
+
+### MoH Example 1
```
```suggestion
### CHT Deployment Example 1
```

andrablaj commented on content/en/hosting/kubernetes-vs-docker.md:
```diff
@@ -2,48 +2,118 @@
+
+### MoH Example 1
+
+This MoH hosts 47 production instances in a national data center running [K3s](https://k3s.io/). They run a simplified k3s set up with 47 isolated installations of k3s to easy management complexity.  Thus each CHT instance is bound to the VM's local storage.  If a VM fails, there is no automatic fail-over and a restore from backup is needed.
```
```suggestion
This deployment hosts 47 production instances in a data center running [K3s](https://k3s.io/). It runs a simplified k3s set up with 47 isolated installations of k3s to manage complexity easily.  Thus each CHT instance is bound to the VM's local storage.  If a VM fails, there is no automatic failover, and a restore from backup is needed.
```

andrablaj commented on content/en/hosting/kubernetes-vs-docker.md:
```diff
@@ -2,48 +2,118 @@
+
+
+
+### MoH Example 2
```
```suggestion
### CHT Deployment Example 2
```

andrablaj commented on content/en/hosting/kubernetes-vs-docker.md:
```diff
@@ -2,48 +2,118 @@
+
+### MoH Example 2
+
+This MoH hosts 1 of production instance in a national data center running [K3s](https://k3s.io/).
```
```suggestion
This deployment hosts 1 CHT production instance in a data center running [K3s](https://k3s.io/).
```

andrablaj commented on content/en/hosting/kubernetes-vs-docker.md:
```diff
@@ -2,48 +2,118 @@
+|            Users: | 18k                       |
+
+
+### Medic
```
```suggestion
### CHT Deployment Example 3
```

andrablaj commented on content/en/hosting/kubernetes-vs-docker.md:
```diff
@@ -2,48 +2,118 @@
+
+### Medic
+
+Medic hosts 46 production instances on Amazon [Elastic Kubernetes Service](https://docs.aws.amazon.com/eks/latest/userguide/what-is-eks.html) (EKS).
```
```suggestion
This deployment hosts 46 production instances on Amazon [Elastic Kubernetes Service](https://docs.aws.amazon.com/eks/latest/userguide/what-is-eks.html) (EKS).
```

andrablaj reviewed (changes requested):
Almost there! 💛

I added extra anonymisation of deployments per Slack conversation. The rest of the comments are non-blocking/nitpicks. 

@dianabarsan @Hareet I would really love you to review this PR. My review was strongly focused on wording, and it needs extra eyes from a technical/numbers perspective. Thank you!

dianabarsan commented on content/en/hosting/kubernetes-vs-docker.md:
```diff
@@ -2,48 +2,118 @@
+
+|                   |                             |
+|------------------:|:----------------------------|
+|    CHT Instances: | 47                          |
```
I think the more important breakdown here was by instance, and how many users / resources / repication apdex they have, at least that's what I was envisioning.

mrjones-plip replied:
Do you mean for all three examples, or for just Medic hosted examples?  Assuming just medic hosted, maybe breaking out two examples deployments, one small and one big?

mrjones-plip commented:
Yay!  Thanks for iterating with me on this @andrablaj  - I think your input was really valuable to distill this down.

I'll wait to work through @dianabarsan's feedback on the example instance specifics before merging.

dianabarsan commented on content/en/hosting/kubernetes-vs-docker.md:
```diff
@@ -2,48 +2,124 @@
+## Example deployments
 
-Like Kubernetes above, Docker Compose deploys the same services but adds an additional 7th to allow for ingress/egress via a reverse proxy:
+Below are examples from real world large-scale CHT deployments which have been anonymised.  Users are active users for the past 30 days.
```
```suggestion
Below are examples from real world large-scale CHT deployments which have been anonymized. User counts represent the number of active users within a 30 days interval. 
```

dianabarsan reviewed (approved):
Cool. I have have wording suggestion.

andrablaj commented on content/en/hosting/4.x/_index.md:
```diff
@@ -12,15 +12,16 @@ aliases:
   {{< card link="app-developer" title="App Developer Hosting" subtitle="Learn how to host the CHT when developing apps" icon="server" >}}
   {{< card link="migration/" title="Migration Guides" icon="arrow-circle-right" subtitle="Guides for migrating CHT applications" >}}
-  {{< card link="production/" title="Production Hosting" subtitle="Details for hosting the CHT on self run infrastructure" icon="template" >}}
+  {{< card link="/hosting/4.x/docker/" title="Production Docker" subtitle="Details for hosting the CHT on Docker" icon="docker" >}}
```
How about going an extra level of coolness and marking the Docker card as the recommended path, now that we have [cool Hextra cards](https://imfing.github.io/hextra/docs/guide/shortcodes/cards/)? 😎
